doc/raspbian_install.md

```diff
@@ -0,0 +1,130 @@
+
+These are the steps I followed to create a _local_, _non-federated_ dev server for goblin.
+
+## Environment
+
+* Raspbian 11 (bullseye) on a Raspberry Pi 4 (4gb memory)
+* Node20.x (follow a guide like https://xavier.arnaus.net/blog/install-nodejs-20-into-a-raspberry-pi-4)
+* Local redis (installed via apt)
+* Local postgres (installed via apt)
+
+## A note about shell commands
+
+Each block of shell commands is meant to be stand-alone. If you want to just do everything as the `goblin` user (which we're going to create in a moment) you can omit the `sudo -u goblin bash` lines.
+
+## Storage
+
+You'll need several gigs of free storage to install everything goblin needs to get running.
+
+You'll also hate yourself if, like me, you are dumb enough to do the install on SD media.
+
+An external SSD and an enclosure for it costs less money than you'd spend buying combo meals for four at McDonald's. (Do not try to eat the SSD, no matter how much ketchup you put on it.)
+
+## Dependencies
+
+Here's a partial list of apt dependencies to get things up and running. It doesn't include dependencies that I already had installed thanks to other things running on the . (Ask me how much I want to spin up a completely new Raspbian install to verify this.)
+
+```sh
+sudo apt install gcc-core g++ libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libnss3 libxss1 libasound2 libxtst6 xauth xvfb redis postgresql nginx
+npm i -g pnpm
+```
+
+## Create user and clone repository
+
+We're going to create a special `goblin` user to isolate the goblin install. If you're asked for its home directory, specify it as `/home/goblin`.
+
+We are going to install the app into `/etc/goblin` to isolate it from the rest of the system. (It'll be served over nginx when you do things for real.)
+
+```sh
+sudo adduser goblin
+sudo bash
+cd /etc
+sudo git clone https://github.com/johnHackworth/goblin.git
+sudo chown -R goblin:goblin goblin
+```
+
+## Getting pnpm running
+
+The first thing you'll need to do is get `corepack` up and running so that it can get `pnpm` up and running. We'll install `corepack` underneath `/etc/goblin`.
+
+Note that `pnpm` also needs to do some `npm` stuff, and you'll need to be able to write to a `.npm` directory in $HOME, so let's get that out of the way first:
+
+```sh
+sudo -u goblin bash
+cd /etc/goblin
+corepack use pnpm@8
+```
+
+If you are trying to install to a directory that's on an SD card, go get several cups of coffee and a fun YouTube playlist. You'll be here a while waiting for `corepack use` to finish.
+
+...and then you will find the install fails halfway through! Cypress thinks that you don't satisfy its system requirements. (The Cypress devs do not think people want to use it on a Pi, so their install script doesn't recognize the `armv7` architecture. Their install script defaults to assuming any architecture it doesn't recognize is ia32.) You must fool it. Go into `node_modules/cypress/lib/util.js` and edit the `getRealArch()` function:
```

Comment:
didn't we remove Cypress already? or it's still there somewhere

Reply:
Not yet, apparently:
https://github.com/johnHackworth/goblin/blob/4a116a8f606d110a9b79e4315eb7de7d1f360cdc/pnpm-lock.yaml#L48

On the other hand pnpm *does* think it's gone, because `pnpm remove -D cypress` keeps telling me it isn't in devDependencies:

 ERR_PNPM_CANNOT_REMOVE_MISSING_DEPS  Cannot remove 'cypress': no such dependency found in 'devDependencies'

Available dependencies: @types/gulp, @types/gulp-rename, @types/node, chalk, cross-env, execa, gulp, gulp-cssnano, gulp-rename, gulp-replace, gulp-terser, install-peers, rome, start-server-and-test, typescript